executor/aggfuncs/aggfuncs.go

```diff
@@ -45,6 +45,11 @@ var (
 	_ AggFunc = (*approxCountDistinctPartial2)(nil)
 	_ AggFunc = (*approxCountDistinctFinal)(nil)
 
+	// All the AggFunc implementations for "APPROX_PERCENTILE" are listed here.
```

Comment:
1. Percentile of deduplicated data set looks weird. In my opinion, we should ignore it, or treat it as an error.
2. Yes, I haven't find a way to calculate exact percentile value without a global view of data set. We can also do parallel partial sorts and record count for each partition, then merge them in final phase, which can hardly better than just collect data and do selection in final phase.

Besides, I found that there do exist some methods to approximate percentile of a data stream, which can be faster and consume less memory. Thus, `PERCENTILE` or `PERCENTILE_EXACT` seems to be a better name for this function.

Reply:
if without groupby, it can be optimized into two steps:
partical step: return the top Percentile, and total count of this part;
final step: get the p-th value.

Reply:
the partical step can be pushed down.